dao/src/main/java/org/thingsboard/server/dao/ai/AiModelServiceImpl.java

```diff
@@ -103,10 +117,19 @@ public FluentFuture<Optional<AiModel>> findAiModelByTenantIdAndIdAsync(TenantId
         return FluentFuture.from(jpaExecutor.submit(() -> findAiModelByTenantIdAndId(tenantId, modelId)));
     }
 
+    @Override
+    public Optional<AiModel> findAiModelByTenantIdAndName(TenantId tenantId, String name) {
+        return Optional.ofNullable(aiModelDao.findByTenantIdAndName(tenantId.getId(), name));
+    }
+
     @Override
     @Transactional
     public boolean deleteByTenantIdAndId(TenantId tenantId, AiModelId modelId) {
-        return deleteByTenantIdAndIdInternal(tenantId, modelId);
+        AiModel aiModel = aiModelDao.findById(tenantId, modelId.getId());
+        if (aiModel == null) {
```

Comment:
Could you clarify why we need this check?
What exactly does the method
`aiModelDao.deleteByTenantIdAndId(tenantId, aiModel.getId());`
return if the model with this ID doesn’t exist?

If the check is indeed necessary, it might be better to move it into `deleteByTenantIdAndIdInternal` to avoid duplicating logic in both `deleteEntity` and `deleteByTenantIdAndId`.